examples/sealed-auctions/src/types.rs

```diff
@@ -0,0 +1,83 @@
+extern crate chrono;
+
+use std::{collections::HashSet, vec::Vec}; // `use` statements can go anywhere
+
+pub type UserId = mantle::types::Address;
+pub type ItemId = usize;
+
+// A bid struct with the bidder and latest bid
+#[derive(Serialize, Deserialize, Clone, Debug, Eq)]
+pub struct Bid {
+    pub bidder: UserId,
+    pub value: usize,
+}
+
+impl PartialEq for Bid {
+    fn eq(&self, other: &Bid) -> bool {
+        self.bidder == other.bidder
+    }
+}
+
+impl PartialOrd for Bid {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.value.partial_cmp(&other.value)
+    }
+}
+
+impl std::hash::Hash for Bid {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.bidder.hash(state);
+    }
+}
+
+impl std::borrow::Borrow<UserId> for Bid {
+    fn borrow(&self) -> &UserId {
+        &self.bidder
+    }
+}
+
+// The auction struct for every auction that has taken place on the platform
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct Auction {
+    pub item_id: ItemId,
+    pub bids: HashSet<Bid>,
```

Comment:
Are you sure that you don't actually want a `HashMap`?

Reply:
Thought of that but then decided against it as the HashSet is implemented as a HashMap under the hood is what I read.

Reply:
right, but isn't the API you want key-value? At least to me, it looks like you're using it as a kv store. I'd buy it if a person could have multiple bids, but it looks like you're only using the max bid per person.

Reply:
Done. `HashMap` now.